### PR TITLE
feat: support filtering jobs by user

### PR DESCRIPTION
**Pull request by ryanfang5**

### CODE REVIEW

File: virtool/jobs/api.py [hunk 1]
```diff
@@ -25,7 +25,14 @@
...
 @routes.view("/jobs")
 class JobsView(PydanticView):
-    async def get(self) -> Union[r200[List[GetJobResponse]], r400]:
+    async def get(
+        self,
+        archived: Optional[bool] = None,
+        page: conint(ge=1) = 1,
+        per_page: conint(ge=1, le=100) = 25,
+        state: List[str] = Field(default_factory=list),
+        users: List[str] = Field(default_factory=list),
```
Comment (igboyes) [inline virtool/jobs/api.py:34]:
```suggestion
        user: List[str] = Field(default_factory=list),
```

File: virtool/jobs/api.py [hunk 2]
```diff
@@ -25,7 +25,14 @@
...
+    ) -> Union[r200[List[GetJobResponse]], r400]:
```
Comment (igboyes) [inline virtool/jobs/api.py:35]: Isn't this endpoint paginated? If so, this response model is incorrect.

File: tests/jobs/test_api.py [hunk 3]
```diff
@@ -18,6 +19,10 @@ async def test_find_beta(archived, state, fake, snapshot, spawn_client):
     if state is not None:
         url += f"&state={state}"
 
+    if users is not None:
+        for user in users.split(","):
```
Comment (igboyes) [inline tests/jobs/test_api.py:23]: Is `split` still necessary here?

**Review by igboyes — CHANGES_REQUESTED** [review]

File: tests/jobs/test_api.py (hunk 3 above)
Comment (ryanfang5) [reply tests/jobs/test_api.py:23]: Not sure why I kept this, I have changed the parametrization to pass in a list of users.

**Review by igboyes — APPROVED** [review]

**Pull request merged**